### ISSUE TITLE
pad_id not in nn.Embedding vocab

### ISSUE THREAD

**Issue by eonglints** [lucidrains/audiolm-pytorch#18]
Hey, I'm sure you're aware of this, but flagging just in case.

As you know, in the `SemanticTransformer`, the [`pad_id` defaults to `-1`](https://github.com/lucidrains/audiolm-pytorch/blob/e0a4fd944024680886819d6d603bc4e6110ad492/audiolm_pytorch/audiolm_pytorch.py#L323). When using a batch size greater than 1, tokens are padded by [`batch_unique_consecutive`](https://github.com/lucidrains/audiolm-pytorch/blob/e0a4fd944024680886819d6d603bc4e6110ad492/audiolm_pytorch/audiolm_pytorch.py#L768), using this `pad_id`. 

This then causes an issue when these padded token sequences are [passed through the embedding layer](https://github.com/lucidrains/audiolm-pytorch/blob/e0a4fd944024680886819d6d603bc4e6110ad492/audiolm_pytorch/audiolm_pytorch.py#L392) as the embedding layer has [been initialized](https://github.com/lucidrains/audiolm-pytorch/blob/e0a4fd944024680886819d6d603bc4e6110ad492/audiolm_pytorch/audiolm_pytorch.py#L335) with a vocab size of `num_semantic_tokens + 1`, which doesn't include this default `pad_id`. This obviously causes an index out of range error (which can be tricky to spot immediately when running with CUDA, easier to spot on CPU).

Perhaps the `pad_id` should be `self.eos_id + 1` (equivalent to `num_semantic_tokens + 2`)? Or maybe there's a smarter solution...

**Comment by lucidrains** [comment]
@eonglints hey! yea, the use of `-1` as padding comes with its pros and cons

mainly i wanted to have one to one correspondence with the original codebook ids

i've added a helper, function `get_embeds`, that should help!

ex.

```python
embeds = get_embeds(semantic_transformer.semantic_embedding, sample)
```

**Comment by eonglints** [comment]
Beautiful! That's a very elegant solution, thanks.
